File: fix-modal.js
const fs = require('fs');

let content = fs.readFileSync('src/pages/index/index.tsx', 'utf8');
const lines = content.split('\n');
const newLines = [];
let inDuplicateModal = false;
let skipCount = 0;

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    
    // 检查是否是重复的内联模态框（在文件末尾之前）
    // 保留最后使用 LoginModal 组件的部分
    if (i < lines.length - 50 && line.includes('showLoginModal &&')) {
        // 检查下一行是否包含 login-modal-mask
        if (lines[i + 1] && lines[i + 1].includes('login-modal-mask')) {
            inDuplicateModal = true;
            skipCount = 1;
            continue;
        }
    }
    
    if (inDuplicateModal) {
        // 计算 View 标签的嵌套层级
        const openTags = (line.match(/<View/g) || []).length;
        const closeTags = (line.match(/<\/View>/g) || []).length;
        skipCount += openTags - closeTags;
        
        if (skipCount <= 0) {
            inDuplicateModal = false;
        }
        continue;
    }
    
    newLines.push(line);
}

fs.writeFileSync('src/pages/index/index.tsx', newLines.join('\n'), 'utf8');
console.log('✅ 已删除所有重复的内联登录模态框代码');
console.log('✅ 保留了文件末尾的 LoginModal 和 WechatLogin 组件引用');

File: fix_modal.py
# 读取文件
with open('src/pages/index/index.tsx', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# 过滤掉包含特定关键词的行
new_lines = []
skip_mode = False
skip_count = 0

for i, line in enumerate(lines):
    # 检查是否需要开始跳过
    if 'login-modal-mask' in line or ('login-modal-content' in line and '完善个人资料' in ''.join(lines[i:i+5])):
        skip_mode = True
        skip_count += 1
        print(f'开始跳过代码块 #{skip_count} 在行 {i+1}')
    
    if skip_mode:
        # 检查是否结束跳过
        if ')}' in line and i > 0 and ')' in lines[i-1]:
            skip_mode = False
            print(f'结束跳过代码块 #{skip_count}')
            continue
        # 跳过这一行
        continue
    else:
        new_lines.append(line)

# 写回文件
with open('src/pages/index/index.tsx', 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print(f'已完成清理，共删除 {skip_count} 个代码块')
